Automattic/jetpack
# Add babel-plugin-replace-textdomain

#### Changes proposed in this Pull Request:

This is a (surprisingly simple) Babel plugin intended for use when
building Webpack bundles using Jetpack RNA components, so the resulting
bundles will use the plugin's textdomain (since that's where WordPress's
translation infrastructure will put the strings) instead of each
components'.

This also adds it as an option to our webpack-config package, but
does not enable it anywhere yet.

#### Jetpack product discussion


PT: p9dueE-3MG-p2

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Is CI happy?
* Try the following:
  1. `jetpack build plugins/backup`
  2. Look at the built `projects/plugins/backup/build/index.js`. Note that many calls to `__()` supply "jetpack" as the textdomain, even though the plugin's textdomain is "jetpack-backup".
  3. Edit  `projects/plugins/backup/babel.config.js` to look like
     ```js
     const config = {
         presets: [
             [ '@automattic/jetpack-webpack-config/babel/preset', { pluginReplaceTextdomain: { textdomain: 'jetpack-backup' } } ],
         ],
     };
     
     module.exports = config;
     ```
  4. `jetpack build plugins/backup` again.
  5. Look at the built `projects/plugins/backup/build/index.js`. The calls to `__()` should now all supply "jetpack-backup" as the textdomain.
